### PR TITLE
Makes Description Optional on Personal Schedule

### LINKED ISSUE #8: Make Description Optional on Personal Schedule
Currently, the description field on personal schedule is required.

Make it optional.

### PR DESCRIPTION
**Pull request by bsaiju**
Closes #8 

# Before commit:
## When creating personal schedule without description:
Screenshot 2024-11-16 at 3 43 06 AM

## When editing personal schedule to clear out description:
Screenshot 2024-11-16 at 3 43 49 AM

# After commit:
## When creating personal schedule without description:
Screenshot 2024-11-16 at 3 48 51 AM
*Clicks Create Button*
Screenshot 2024-11-16 at 3 49 09 AM

## When editing personal schedule to clear out description:
Screenshot 2024-11-16 at 3 50 09 AM
*Clicks Edit button*
Screenshot 2024-11-16 at 3 50 28 AM
*Clicks Update Button*
Screenshot 2024-11-16 at 3 50 53 AM

# Dokku Link: [https://courses-bsaiju.dokku-03.cs.ucsb.edu/](https://courses-bsaiju.dokku-03.cs.ucsb.edu/)

### CODE REVIEW

**Review by pconrad — CHANGES_REQUESTED** [review]
I have a question about whether adding `stryker` to package.json is necessary.

Further, if it is, is probably doesn't go in `dependencies` but in `dev-dependencies`, but still, I would have expected that `"@stryker-mutator/jest-runner": "^8.5.0",` in `dev-dependencies` would be enough.

File: frontend/src/tests/components/PersonalSchedules/PersonalScheduleForm.test.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ describe("PersonalScheduleForm tests", () => {
     fireEvent.click(submitButton);
 
     expect(await screen.findByText(/Name is required./)).toBeInTheDocument();
-    expect(screen.getByText(/Description is required./)).toBeInTheDocument();
+    // expect(screen.getByText(/Description is required./)).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (bsaiju) [reply frontend/src/tests/components/PersonalSchedules/PersonalScheduleForm.test.js]: this is fixed in the previous commit.

**Comment by bsaiju** [comment]
> I have a question about whether adding `stryker` to package.json is necessary.
> 
> Further, if it is, is probably doesn't go in `dependencies` but in `dev-dependencies`, but still, I would have expected that `"@stryker-mutator/jest-runner": "^8.5.0",` in `dev-dependencies` would be enough.

I did not specifically edit it. I believe it was added when I ran npm install or mvn install or something of the like. Should I delete that line?

**Review by pconrad — CHANGES_REQUESTED** [review]
Please restore package-lock.json since package.json is not being changed.

https://ucsb-cs156.github.io/topics/pull_requests/package_lock_json.html

**Review by ChristopherChau — DISMISSED** [review]
LGTM

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove changes to package-lock.json from this PR.

Please see: https://ucsb-cs156.github.io/topics/pull_requests/package_lock_json.html

**Comment by pconrad** [comment]
When I said: remove the changes to package-lock.json, I meant it.

image

This still shows that package-lock.json was updated.

You need to follow the steps at this page: https://ucsb-cs156.github.io/topics/pull_requests/package_lock_json.html

When you are done, the ONLY changes to this PR should be in two files:
* frontend/src/main/components/PersonalSchedules/PersonalScheduleForm.js
* frontend/src/tests/components/PersonalSchedules/PersonalScheduleForm.test.js

If you can't figure out how to do that in this branch, then close this PR, make a new branch off of main, and reapply the simple 1-2 lines changes in a new PR.  I said that I would not merge this one with change to package-lock.json, and I still won't.

If you want the points, then address the changes in the code review.  Don't just "remove the label".  That wastes everyone's time.

**Pull request closed without merging**